### PR TITLE
yofi: fix build

### PR DESCRIPTION
**Pull request by ocfox**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/yo/yofi/Cargo.lock
Comment (getchoo) [inline pkgs/by-name/yo/yofi/Cargo.lock:1]: As it seems this project doesn't use git dependencies, we should be able to use a patch generated by `cargo update time` as opposed to vendoring the entire lockfile. You can find an example of this in https://github.com/NixOS/nixpkgs/pull/334764

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/yo/yofi/Cargo.lock
Comment (ocfox) [reply pkgs/by-name/yo/yofi/Cargo.lock:1]: Thanks, learned `cargoPatches`.

**Review by getchoo — APPROVED** [review]

**Comment by RaySlash** [comment]
Hey @ocfox , I believe it would be easier to follow the upstream repo changes rather than making a patch yourselves. You can update the rev of yofi itself as the `time` crate issue has been fixed upstream. As there is no official release, you could name it `unstable-05102024` for example.

**Comment by ocfox** [comment]
Maybe patch cargo.lock more cleaner?

**Review by getchoo — CHANGES_REQUESTED** [review]
Ah didn't realize this was a patch from a fork; we probably shouldn't be using it then

https://github.com/l4l/yofi/commit/09901e75cbdf2147553ab888adde480e57baa0d1 would be a better candidate. I'd recommend continuing to pull it in as a patch though, since unstable versions can introduce issues we don't intend

**Comment by ocfox** [comment]
> [l4l/yofi@09901e7](https://github.com/l4l/yofi/commit/09901e75cbdf2147553ab888adde480e57baa0d1) would be a better candidate. I'd recommend continuing to pull it in as a patch though, since unstable versions can introduce issues we don't intend

I just realized that I tried, because of `flake.nix` caused the patch to fail, so I used a fork commit.

**Pull request merged**